experimental/sql/dialects/ibis_dialect.py

```diff
@@ -0,0 +1,247 @@
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from dataclasses import dataclass
+from typing import Any
+from xdsl.ir import Block, Region, Operation, SSAValue, ParametrizedAttribute, Data, MLContext, Attribute
+from xdsl.dialects.builtin import StringAttr, ArrayAttr, ArrayOfConstraint, IntegerAttr
+from xdsl.irdl import AttributeDef, OperandDef, ResultDef, RegionDef, SingleBlockRegionDef, irdl_attr_definition, irdl_op_definition, ParameterDef, AnyAttr, VarOperandDef, builder
+
+# This file defines a dialect that mirrors the internal data
+# structures of the ibis framework (https://ibis-project.org/docs/dev/).
+# DataTypes are defined as Attributes, other important nodes are
+# operations. Everything that is an `expresssion` in ibis is not
+# modelled.
```

Comment:
I think I would elaborate a bit here:
* Describe the fact that the ibis query is sort of a tree but not really and that that's an artifact of the UI.
* Describe the expression vs operation weirdness in Ibis and that that's an artifact of the UI as well.
* Say that this dialect should represent a cleaned-up version of that tree and what exactly it cleans up.

Reply:
Also, I can't really match "other important nodes are operations" with the structure of the IR... Aren't they (single-op) regions?

Reply:
This, plus I'd also mention that we model a tree where every op is a single region terminator and without any SSA value in this dialect atm.